cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/ConfigManager.kt

```diff
@@ -62,18 +65,38 @@ class ConfigManager : AbstractVerticle() {
     private fun attachEndpoint(config: JsonObject) {
         val endpointId = config.getString(CogboardConstants.PROP_ENDPOINT)
         endpointId?.let {
-            config.put(CogboardConstants.PROP_ENDPOINT, endpoints
-                    .stream()
-                    .map { it as JsonObject }
-                    .filter {
-                        endpointId == it.getString("id")
-                    }.findFirst()
-                    .orElse(JsonObject()))
+            config.put(CogboardConstants.PROP_ENDPOINT,
+                    findJsonObjectById(endpointId, endpoints)
+                            .map { attachCredentials(it) }
+                            .orElse(JsonObject()))
         }
     }
 
+    private fun attachCredentials(endpoint: JsonObject): JsonObject {
+        endpoint.remove(CREDENTIALS).toString().let {
+            val credentials = findJsonObjectById(it, credentials)
+                    .orElse(JsonObject())
+            endpoint.put(USER, credentials.getString(USER))
+            endpoint.put(PASSWORD, credentials.getString(PASSWORD))
+        }
+        return endpoint
+    }
+
+    private fun findJsonObjectById(id: String?, array: JsonArray): Optional<JsonObject> {
```

Comment:
You can use **Extension Functions** this.
https://kotlinlang.org/docs/reference/extensions.html#extension-functions